eng/tools/suppressions/test/e2e.test.ts

```diff
@@ -0,0 +1,38 @@
+import { expect, test } from "vitest";
+import { getSuppressions } from "../src/suppressions.js";
+import { join } from "path";
+
+async function getTestSuppressions(path: string, tool: string = "TestTool") {
+  return await getSuppressions(tool, join(__dirname, "e2e", path));
+}
+
+test.concurrent("no suppressions.yaml", async () => {
+  const suppressions = await getTestSuppressions("noSuppressionsYaml");
+  expect(suppressions).toEqual([]);
+});
+
+test.concurrent("empty suppressions.yaml", async () => {
+  const suppressions = await getTestSuppressions("emptySuppressionsYaml");
+  expect(suppressions).toEqual([]);
+});
+
+test.concurrent("suppress foo.json, get foo.json", async () => {
+  const suppressions = await getTestSuppressions(join("suppressFooJson", "foo.json"));
+  expect(suppressions).toEqual([
+    {
+      tool: "TestTool",
+      paths: ["foo.json"],
+      reason: "test",
+    },
+  ]);
+});
+
+test.concurrent("suppress foo.json, get bar.json", async () => {
```

Comment:
I feel there is a lite bit too large of mental shortcut here right now. I look at this test and I am unsure what it means. I guess it is something like: 

> Given having only 1 suppression file, that has a suppression for something in `foo.json` path, when I am asking for all suppressions of `bar.json`, I get empty result

Following test contents would be more readable to me (see my comment about using fixtures):

``` typescript
  const suppressionFiles = suppressionFixture.fileWithFooJsonPathSuppression
  
  // Act
  const suppressions = getSuppressionsForFile(suppressionFile: suppressionFiles, targetPath: "bar.json")

  expect(suppressions).toEqual([]);
```

or even 1-liner:

``` typescript
expect(suppressionsFor(suppressionFixture.fileWithFooJsonPathSuppression, targetPath: "bar.json").toEqual([])

```

and to be super-fancy and compact you could do a declarative test:

``` typescript

test("suppression scenarios", async () => {

  const f = suppressionsFixture
  const testCases = [
    [f.fileWithFooJsonPathSuppression, "foo.json" , f.fooJsonSuppressionEntry]
    [f.fileWithFooJsonPathSuppression, "bar.json" , []]
    [f.fileWithBarJsonPathSuppression, "bar.json" , f.barJsonSuppressionEntry]
    ...
  ]
  assertIndependentlyAndInParallel(testCases);
})
```

or even more compact:

``` typescript

test("suppression scenarios", async () => {
  assertIndependentlyAndInParallel((f: fixture) => [
    [f.fileWithFooJsonPathSuppression, "foo.json" , f.fooJsonSuppressionEntry]
    [f.fileWithFooJsonPathSuppression, "bar.json" , []]
    [f.fileWithBarJsonPathSuppression, "bar.json" , f.barJsonSuppressionEntry]
    ...
  ]);
})
```

I am fine either way. Just some ideas.

Reply:
I can improve the name to be more descriptive, but I think the tests should be pretty clear as written.  The test should make sense, with the context of the "spec folder" under `suppressFooJson`.  You can see the folder has two files `bar.json` and `foo.json`, but the `suppressions.yaml` only has a suppression for `foo.json`.

These were just quick sanity tests, to make sure the tool works at a basic level of file selection.

Reply:
To be specific, this threw me off a bit:

`await getTestSuppressions(join("suppressFooJson", "bar.json"));`

If this was a bit more declarative I would have less trouble. I was just a bit confused why we are joining these two (because of fixture file name, which is an infra detail, not test essence) and it took me a moment to realize that `FooJson` and `bar.json` both reference to path - one a path in the suppressions file, and the other one the desired data.

Reply:
This line:

```
await getTestSuppressions(join("suppressFooJson", "bar.json"));
```

Means:

Use the `getSuppressions()` API, to load suppressions for file `/root/repo/eng/suppressions/test/e2e/suppressFooJson/bar.json`, for tool "TestTool"

Reply:
I think if we would add a comment somewhere in the file explaining this pattern interpretation that would be helpful and totally enough.